src/apple-other.rs

```diff
@@ -14,7 +14,7 @@ extern "C" {
 }
 
 pub fn getrandom_inner(dest: &mut [MaybeUninit<u8>]) -> Result<(), Error> {
-    let ret = unsafe { CCRandomGenerateBytes(dest.as_mut_ptr() as *mut c_void, dest.len()) };
+    let ret = unsafe { CCRandomGenerateBytes(dest.as_mut_ptr().cast::<c_void>(), dest.len()) };
```

Comment:
I think we can rely on type inference and use just `.cast()`.

Reply:
I intentionally avoid using type inference so that we ensure we're casting to the expected type. That is one of the goals of this change.

Reply:
I don't think it helps in any practical way in this case. In the first place, pointer type does no matter on the ABI level, so we are doing casting to just satisfy signature of external functions. Also, IIRC `std` also uses a lot of pointer casts which rely on type inference in cases like this.

Reply:
It doesn't seem to help in our code because our code does seem to be correct. But this style of explicitly naming the type has helped prevent bugs in practice in other code, and it makes the code easier to audit (from my own experience).

Reply:
@josephlr WDYT?

Reply:
I personally like having the type explicit in the cast. It makes it easier for me to understand what's going on. It's not a huge deal, but it is nice to confirm that nothing unexpected is happening with the cast.